### ISSUE TITLE
Warning the underscored variable `"_t"` is used after being set

### ISSUE THREAD

**Issue by shahryarjb** [gjaldon/ecto_enum#118]
Hi

### My environment
```elixir
Erlang/OTP 25 [erts-13.0.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]
Elixir 1.14.2 (compiled with Erlang/OTP 25
MacOs 12.4
ecto_enum: 1.4
```

### The warning I get from starting my app
```elixir
warning: the underscored variable "_t" is used after being set. A leading underscore indicates that the value of the variable should be ignored. If this is intended please rename the variable to remove the underscore
  lib/ecto_enum/typespec.ex:26: EctoEnum.Typespec.TestModule

warning: the underscored variable "_t" is used after being set. A leading underscore indicates that the value of the variable should be ignored. If this is intended please rename the variable to remove the underscore
  lib/ecto_enum/typespec.ex:28: EctoEnum.Typespec.TestModule
```

**Comment by ryanwinchester** [comment]
Same

**Comment by IceDragon200** [comment]
I believe it's ~some codegen~ something added in `elixir 1.14.2`. 

`1.14.1` does not have this problem, I've had to rollback to `1.14.1` to get around this for, because it doesn't appear to be a `ecto_enum` problem specifically (at least not one I can visibly identify after looking through the code)

tl;dr

Use `elixir 1.14.1` to have nice and quiet compiles (or to actually compile if you used `warnings_as_errors` like I do)

**Comment by jeremyjh** [comment]
It looks like this is a bug in elixirc that has been fixed on HEAD but not yet released: https://github.com/elixir-lang/elixir/commit/bfa799b31ffff3801bb401d4f6ea3dce5d3c2fd7

**Comment by gjaldon** [comment]
@shahryarjb looks like @jeremyjh is right. It seems like a regression introduced in elixir `1.14.2`. Could you verify that's the case before I close this?

**Comment by shahryarjb** [comment]
Hi @gjaldon , I re-test in elixir 1.14.1 and I had no warning, but I have no access to test it in master or main branch of elixir.

I think we can wait for new version of elixir, and if it won't be fixed, so we can re-open this issue

Thank you in advance.

**Comment by gjaldon** [comment]
@shahryarjb ok great! sounds like a plan :)
